cas-server-core/src/main/java/org/jasig/cas/authentication/principal/WebApplicationServiceFactory.java

```diff
@@ -55,9 +55,12 @@ public WebApplicationService createService(final HttpServletRequest request) {
         final String id = cleanupUrl(serviceToUse);
         final String artifactId = request.getParameter(CasProtocolConstants.PARAMETER_TICKET);
 
-        return new SimpleWebApplicationServiceImpl(id, serviceToUse,
-                artifactId, "POST".equalsIgnoreCase(method) ? Response.ResponseType.POST
-                : Response.ResponseType.REDIRECT);
+        final  Response.ResponseType type = "POST".equalsIgnoreCase(method) ? Response.ResponseType.POST
+                                                        : Response.ResponseType.REDIRECT;
+
+        final WebApplicationService webApplicationService = new SimpleWebApplicationServiceImpl(id, serviceToUse,
+                artifactId, new WebApplicationServiceResponseBuilder(type));
```

Comment:
Do we need a new `ResponseBuilder` for each service? I tend to think the response builder is a singleton (one by type).